nav2_mppi_controller/src/path_handler.cpp

```diff
@@ -67,14 +66,46 @@ PathHandler::getGlobalPlanConsideringBoundsInCostmapFrame(
       return euclidean_distance(global_pose, ps);
     });
 
-  nav_msgs::msg::Path transformed_plan;
-  transformed_plan.header.frame_id = costmap_->getGlobalFrameID();
-  transformed_plan.header.stamp = global_pose.header.stamp;
-
   auto pruned_plan_end =
     nav2_util::geometry_utils::first_after_integrated_distance(
     closest_point, global_plan_up_to_inversion_.poses.end(), prune_distance_);
 
+  return {closest_point, pruned_plan_end};
+}
+
+void PathHandler::calculatePathLengths(
+  const PathIterator & closest_point)
+{
+  // Calculate path length from closest point to end of global_plan_up_to_inversion_
+  auto start_idx_up_to_inversion = std::distance(global_plan_up_to_inversion_.poses.begin(),
+      closest_point);
+  global_plan_length_up_to_inversion_ = nav2_util::geometry_utils::calculate_path_length(
+    global_plan_up_to_inversion_, start_idx_up_to_inversion);
+
+  if (enforce_path_inversion_ && inversion_locale_ != 0u) {
```

Comment:
If we know `enforce_path_inversion_` at this point and it is used in the critics, why have multiple plan length variables for with and without inversion? Shouldn't we need to only store one - perhaps called `local_path_length` to make it clear that this is the local path's length from the robot pose --> local goal. 

I didn't look in detail in this object since the tests are failing, but this only works if we prune paths or know the point on the path that we are located nearest to. Make sure that's considered as some applications don't replan ever so there's not a new path to set the nearest point to the robot as being the zero-th index.